scripts/API.py

```diff
@@ -0,0 +1,418 @@
+# ***************************************************************************
+# * Authors:		Alberto García (alberto.garcia@cnb.csic.es)
+# *
+# *
+# * This program is free software; you can redistribute it and/or modify
+# * it under the terms of the GNU General Public License as published by
+# * the Free Software Foundation; either version 2 of the License, or
+# * (at your option) any later version.
+# *
+# * This program is distributed in the hope that it will be useful,
+# * but WITHOUT ANY WARRANTY; without even the implied warranty of
+# * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# * GNU General Public License for more details.
+# *
+# * You should have received a copy of the GNU General Public License
+# * along with this program; if not, write to the Free Software
+# * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+# * 02111-1307 USA
+# *
+# * All comments concerning this program package may be sent to the
+# * e-mail address 'scipion@cnb.csic.es'
+# ***************************************************************************/
+
+"""
+Module containing all functions needed for the metric's API request.
+"""
+
+# General imports
+import re, hashlib, http.client, json
+
+# Self imports
+from typing import Optional, Tuple
+from .utils import runJob
+
+API_URL = 'xmipp.i2pc.es/api/attempts/'
+
+def sendApiPOST(retCode: int = 0, xmippVersion :str = 'Unknow'):
+	"""
+	### Sends a POST request to Xmipp's metrics's API.
+
+	#### Params:
+	- retCode (int): Optional. Return code for the API request.
+	"""
+	# Getting JSON data for curl command
+	bodyParams = __getJSON(retCode=retCode, xmippVersion=xmippVersion)
+
+	# Send API POST request if there were no errors
+	if bodyParams is not None:
+		# Set up the headers
+		headers = {"Content-type": "application/json"}
+
+		# Establish a connection
+		url = API_URL.split("/", maxsplit=1)
+		path = f"/{url[1]}"
+		url = url[0]
+		conn = http.client.HTTPConnection(url, timeout=4)
+
+		# Send the POST request
+		conn.request("POST", path, bodyParams, headers)
+		response = conn.getresponse()
+		status = response.status
+		# Close the connection
+		conn.close()
+		if status == 200 or status == 201:
```

Comment:
return status == 200 or status == 201

Reply:
Would require importing `http`

Reply:
status == 200 or status == 201 is the same as status == http.HTTPStatus.OK or status == http.HTTPStatus.CREATED
Dont forget that this code will disappear in 4 months...